miniapps/elasticity/materials/neohookean.hpp

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_ELASTICITY_MAT_NEOHOOKEAN_HPP
+#define MFEM_ELASTICITY_MAT_NEOHOOKEAN_HPP
+
+#include "general/enzyme.hpp"
+#include "gradient_type.hpp"
+#include "linalg/tensor.hpp"
+#include "mfem.hpp"
+
+using mfem::internal::tensor;
+using mfem::internal::make_tensor;
+
+/**
+ * @brief Neo-Hookean material
+ *
+ * Defines a Neo-Hookean material response. It satisfies the material_type
+ * interface for ElasticityOperator::SetMaterial. This material type allows
+ * choosing the method of derivative calculation in `action_of_gradient`.
+ * Choices include methods derived by hand using symbolic calculation and a
+ * variety of automatically computed gradient applications, like
+ * - Enzyme forward mode
+ * - Enzyme reverse mode
+ * - Dual number type forward mode
+ * - Finite difference mode
+ *
+ * @tparam dim
+ * @tparam gradient_type
+ */
+template <int dim = 3, GradientType gradient_type = GradientType::Symbolic>
+struct NeoHookeanMaterial
+{
+   static_assert(dim == 3, "NeoHookean model only defined in 3D");
```

Comment:
One can define it for plane stress without any problems - just add the thickness of the slab as parameter.

Reply:
This just indicates that the implementation as is does only work in 3D.

Reply:
Maybe to clarify, say "currently implemented only in 3D"?